GVFS/GVFS.Common/ProductUpgrader.cs

```diff
@@ -154,20 +173,7 @@ public virtual bool TrySetupToolsDirectory(out string upgraderToolPath, out stri
             string rootDirectoryPath = ProductUpgraderInfo.GetUpgradesDirectoryPath();
             string toolsDirectoryPath = Path.Combine(rootDirectoryPath, ToolsDirectory);
 
-            Exception deleteDirectoryException;
-            if (this.fileSystem.DirectoryExists(toolsDirectoryPath) &&
-                !this.fileSystem.TryDeleteDirectory(toolsDirectoryPath, out deleteDirectoryException))
-            {
-                upgraderToolPath = null;
-                error = $"Failed to delete {toolsDirectoryPath} - {deleteDirectoryException.Message}";
-                this.TraceException(deleteDirectoryException, nameof(this.TrySetupToolsDirectory), $"Error deleting {toolsDirectoryPath}.");
-                return false;
-            }
-
-            if (!this.fileSystem.TryCreateOrUpdateDirectoryToAdminModifyPermissions(
-                    this.tracer,
-                    toolsDirectoryPath,
-                    out error))
+            if (!this.productUpgraderPlatformStrategy.TryPrepareApplicationDirectory(out error))
             {
                 upgraderToolPath = null;

```

Comment:
Not directly related to this PR. Can you take a look at lines #182 through #186. `TrySetupToolsDirectory` is trying to copy everything inside `C:\Program Files\GVFS` to "C:\ProgramData\GVFS\GVFS.Upgrade\Tools". The whole set of `GVFS` assemblies get copied. This is not the intent (we only need GVFS.Upgrade.exe and its dependencies), right?

Reply:
> The whole set of GVFS assemblies get copied. This is not the intent (we only need GVFS.Upgrade.exe and its dependencies), right?

The problem was that it was fragile to keep the list of dependencies straight (across all platforms). We now copy over the entire "application", including some extra files, but I don't think there should be an issue with copying over these extra file... If there are concerns with copying the extra files, please let me know :)

Reply:
I see...